### PR TITLE
Starter content removal

### LINKED ISSUE #805: Settings: Add buttons to import and delete Starter content
**Is your feature request related to a problem? Please describe.**
At the moment, the only way to get some demo content is during the last step of the onboarding, "Starter Content", which is an optional step.

**Describe the solution you'd like**
As a first step to the new onboarding, I'd like to have a button in the Footer (when Newspack is in debug mode) to import some demo content (and if content has been imported, we display a second button, to delete the demo content)

**Additional context**
As I said, this is a first step towards the new onboarding where "Starter Content" will be automatically imported when the user clicks on the "Get started" button from the Welcome Screen for the first time. They will also have the possibility to delete the demo content at the end of the onboarding, before being redirected to their homepage or newspack dashboard.

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Starter content removal.
Closes #805

### How to test the changes in this Pull Request:

1. Visit the Setup wizard on a fresh site (`/wp-admin/admin.php?page=newspack-setup-wizard#/`)
2. After completing the setup, observe a "Remove Starter Content" link at the bottom of the Newspack dashboard; click it
3. Observe that the starter content has been removed

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by dkoo — CHANGES_REQUESTED** [review]
Works perfectly. I have a couple of suggestions:

* Since this is a destructive action, should we make the user confirm that they actually want to proceed? e.g. click "Remove starter content," the browser shows a system prompt saying "All Newspack starter posts, pages, and categories will be deleted. Are you sure you want to proceed?" [OK / Cancel]
* After clicking the button, you get dumped back to the Newspack dashboard page with no visual indication that anything has happened. Maybe we should show a notice at the top of the page here that indicates success ("Starter content deleted.") or failure (if any errors cause some starter content to not be fully deleted).

**Comment by adekbadek** [comment]
> Since this is a destructive action, should we make the user confirm that they actually want to proceed? 

I don't think that's necessary, this content is meant to be short-lived.

> After clicking the button, you get dumped back to the Newspack dashboard page with no visual indication that anything has happened.

Added a notice in 2152aa74ba135525f1450187368cec17426a0b24

**Comment by dkoo** [comment]
Hmm, after updating my branch, I'm getting the following JS error whenever I visit the dashboard: 

```
react-dom.min.js?ver=16.13.1:125 TypeError: Object(...) is not a function
    at GlobalNotices (index.js:20)
```

This is true even after re-running `npm install` in the repo root and in `src/components`.

**Comment by adekbadek** [comment]
> Hmm, after updating my branch, I'm getting the following JS error whenever I visit the dashboard:

Can't reproduce, can you set up an ephemeral site with the bug occurring?

**Comment by adekbadek** [comment]
@dkoo Thanks for catching that, should be fixed by d36154b917c7912fda117c16067469fb0f7fa33a

File: includes/class-newspack.php [hunk 1]
```diff
@@ -134,25 +134,36 @@ class_exists( 'Newspack_Popups' ) && 'edit.php?post_type=' . \Newspack_Popups::N
...
+		$redirect_url   = admin_url( 'admin.php?page=newspack' );
 		$newspack_reset = filter_input( INPUT_GET, 'newspack_reset', FILTER_SANITIZE_STRING );
-		if ( 'reset' === $newspack_reset ) {
-			$all_options = wp_load_alloptions();
-			foreach ( $all_options as $key => $value ) {
-				if ( strpos( $key, 'newspack' ) === 0 || strpos( $key, '_newspack' ) === 0 ) {
-					delete_option( $key );
+		if ( 'starter-content' === $newspack_reset ) {
+			Starter_Content::remove_starter_content();
+			$redirect_url = add_query_arg( 'newspack-notice', __( 'Removed Starter Content', 'newspack' ), $redirect_url );
```
Comment (dkoo) [inline includes/class-newspack.php:144]:
```suggestion
			$redirect_url = add_query_arg( 'newspack-notice', __( 'Starter content removed.', 'newspack' ), $redirect_url );
```

**Review by dkoo — APPROVED** [review]
Just a minor nit on the language of the notice, otherwise this is working great now. Feel free to merge whether or not you take the suggestion.

**Pull request merged**